### PR TITLE
Fix for incorrect margin balance and missing websocket callback args

### PR DESCRIPTION
**Pull request by DanishAlsayed**

### CODE REVIEW

File: market_maker/ws/ws_thread.py [hunk 1]
```diff
@@ -318,11 +324,11 @@ def findItemByKeys(keys, table, matchData):
...
             return item
 
+
 if __name__ == "__main__":
     # create console handler and set level to debug
     logger = log.setup_custom_logger('websocket', logging.DEBUG)
     ws = BitMEXWebsocket()
     ws.connect("https://testnet.bitmex.com/api/v1")
-    while(ws.ws.sock.connected):
+    while (ws.ws.sock.connected):
```
Comment (k4r1) [inline market_maker/ws/ws_thread.py:333]: Nit: Please can we remove unrelated formatting changes (if you'd like to make them please do so in a separate PR)

File: market_maker/ws/ws_thread.py [hunk 2]
```diff
@@ -291,16 +296,17 @@ def __on_message(self, message):
...
 
     def __on_close(self):
         logger.info('Websocket Closed')
         self.exit()
 
-    def __on_error(self, error):
+    def __on_error(self, error, arg):
         if not self.exited:
-            self.error(error)
+            self.error(arg)
```
Comment (k4r1) [inline market_maker/ws/ws_thread.py:309]: I'm not sure I understand this change, although I have seen it mentioned in a couple of old issues:
https://github.com/BitMEX/api-connectors/issues/311

It could be to do with how you're running it locally or versions of libs you have - in any case can we please split this out into its own PR since it doesn't look specifically related to the changes you're making here

File: market_maker/bitmex.py [hunk 3]
```diff
@@ -105,9 +105,9 @@ def wrapped(self, *args, **kwargs):
         return wrapped
 
     @authentication_required
-    def funds(self):
+    def funds(self, currency):
```
Comment (k4r1) [inline market_maker/bitmex.py:108]: Nit: It would be better if we could default this second argument to maintain backwards compatibility

**Review by k4r1 — DISMISSED** [review]
Thanks for spending the time on this - I've added some minor comments related to cleanup to make these changes easier to review

File: market_maker/ws/ws_thread.py (hunk 2 above)
Comment (k4r1) [reply market_maker/ws/ws_thread.py:309]: In fact - have you tried [this](https://github.com/BitMEX/sample-market-maker/pull/235/files)?

Comment (DanishAlsayed) [reply market_maker/ws/ws_thread.py]: Thanks. Addressed in 38e8ac6e0ad72214af9609b6a254f23286c3a194

File: market_maker/ws/ws_thread.py (hunk 1 above)
Comment (DanishAlsayed) [reply market_maker/ws/ws_thread.py]: Thanks. Addressed in be0687c6833157dbe5441472e77c0e2532e4f419

File: market_maker/bitmex.py (hunk 3 above)
Comment (DanishAlsayed) [reply market_maker/bitmex.py]: Thanks. Addressed in dc75d1846a9661841e97fd65860a725d6e948dda

File: market_maker/ws/ws_thread.py (hunk 2 above)
Comment (DanishAlsayed) [reply market_maker/ws/ws_thread.py]: > In fact - have you tried [this](https://github.com/BitMEX/sample-market-maker/pull/235/files)?

Yes, didn't help for me.

**Review by byshing — APPROVED** [review]

**Review by byshing — APPROVED** [review]
